### PR TITLE
fix time normalization for az/arcgis

### LINKED ISSUE #357: Fix time normalization in az/arcgis
The normalizer in az/arcgis is erroring on times that should be parsable, likely due to the presence of `am` or `pm`:

```
05/01/2021 22:00:24 WARNING:az/arcgis/normalize.py:unparseable hours: '10:00 AM-12:00 PM'
05/01/2021 22:00:24 WARNING:az/arcgis/normalize.py:unparseable hours: '10:00 AM-12:00 PM'
05/01/2021 22:00:24 WARNING:az/arcgis/normalize.py:unparseable hours: '10:00 AM-12:00 PM'
05/01/2021 22:00:24 WARNING:az/arcgis/normalize.py:unparseable hours: '10:00 AM-12:00 PM'
```

### PR DESCRIPTION
**Pull request by rascalking**
Fix time normalization for AZ arcgis.

| Key Details |
|-|
| Resolves #357 |
State: AZ |
Site: arcgis |

## Before Opening a PR
- [ * ] I tested this using the CLI (e.g., `poetry run vaccine-feed-ingest <state>/<site>`)
- [ * ] I ran auto-formatting: `poetry run tox -e lint-fix`

### CODE REVIEW

File: vaccine_feed_ingest/runners/az/arcgis/normalize.py [hunk 1]
```diff
@@ -145,13 +135,18 @@ def _normalize_hours(
...
     if processed_hours == "8-4":
         return [schema.OpenHour(day=day, opens="08:00", closes="16:00")]
-    if processed_hours == "8:00AM7:00PM":
-        return [schema.OpenHour(day=day, opens="08:00", closes="16:00")]
+    elif processed_hours == "8:00AM7:00PM":
+        return [schema.OpenHour(day=day, opens="08:00", closes="19:00")]
+    elif processed_hours.count("PM") == 2:
+        # "sat_hrs" for multiple Safeway locations are PM to PM, and
+        # "sun_hrs" is the same hours, but AM to PM.  just fix the typo.
+        processed_hours = processed_hours.replace("PM", "AM", 1)
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/az/arcgis/normalize.py:147]: I'm a bit nervous that future updates to the dataset might include validly PM-PM times (e.g., `1:00PM - 8:00PM`).

Could you further limit this `if` to make sure we're updating only the locations with the typo?

Comment (eliblock) [inline vaccine_feed_ingest/runners/az/arcgis/normalize.py:147]: I'm a bit nervous that future updates to the dataset might include validly PM-PM times (e.g., `1:00PM - 8:00PM`).

Could you further limit this `if` to make sure we're updating only the locations with the typo?

File: vaccine_feed_ingest/runners/az/arcgis/normalize.py [hunk 2]
```diff
@@ -168,17 +163,25 @@ def _normalize_hours(
...
 
-    open_time, close_time = processed_hours.split("-")
+    open_time, close_time = [x.strip() for x in re.split(r"\s*-\s*", processed_hours)]
+    opens = _normalize_time(open_time)
+    closes = _normalize_time(close_time)
+
+    # handle the "9-5" case, where the PM for 5 is implied
+    if (opens > closes) and not re.search(r"P\.?M\.?$", close_time):
+        original_closes = datetime.time.fromisoformat(closes)
+        closes = original_closes.replace(hour=original_closes.hour + 12).isoformat()
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/az/arcgis/normalize.py:173]: Good catch!

**Review by eliblock — CHANGES_REQUESTED** [review]

File: vaccine_feed_ingest/runners/az/arcgis/normalize.py (hunk 1 above)
Comment (rascalking) [reply vaccine_feed_ingest/runners/az/arcgis/normalize.py]: Hell, great catch.  Thanks!

**Review by eliblock — APPROVED** [review]

**Pull request merged**